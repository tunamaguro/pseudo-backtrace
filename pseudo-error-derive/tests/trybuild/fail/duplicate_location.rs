use pseudo_error_derive::StackError;

#[derive(Debug, StackError)]
struct DuplicateLocation {
    #[location]
    first: &'static core::panic::Location<'static>,
    #[location]
    second: &'static core::panic::Location<'static>,
}

impl core::fmt::Display for DuplicateLocation {
    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
        write!(f, "duplicate")
    }
}

impl core::error::Error for DuplicateLocation {}

fn main() {}
