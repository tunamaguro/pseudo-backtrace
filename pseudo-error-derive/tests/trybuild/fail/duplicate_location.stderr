error: multiple fields marked with #[location]
 --> tests/trybuild/fail/duplicate_location.rs:7:5
  |
7 |     #[location]
  |     ^
